proposals/lockout-violation-detection.md

```diff
@@ -0,0 +1,38 @@
+### Lockout Violation Detection
+
+We assume a database that we will store vote information ingested from gossip/turbine, which in turn will
+be consulted to detect lockout violations as follows.
+
+1) Track all SlotHashes on the main fork in the database.
```

Comment:
What is the main fork here? How far back to we track - if a validator is slow and votes with an old root would they still be slashed?

Reply:
Would we still need to track pruned forks? Slow validator switching off pruned fork should not be slashed?

Reply:
I think it would be an epoch worth of SlotHashes on the rooted fork. I don't think we care about violations between two side forks/pruned forks, only if there's a violation with the main fork.

So for instance if somebody made a lockout violation on a side fork that conflicts with a vote state on the rooted fork, then we care